libs/pushNotifications.js

```diff
@@ -0,0 +1,108 @@
+var nconf = require('nconf');
+var pushNotify = require('push-notify');
+var Bluebird = require('bluebird');
+var gcmLib = require('node-gcm'); // works with FCM notifications too
+var AWS = require('aws-sdk');
+
+const FCM_API_KEY = nconf.get('PUSH_CONFIGS:FCM_SERVER_API_KEY');
+
+const fcmSender = FCM_API_KEY ? new gcmLib.Sender(FCM_API_KEY) : undefined;
+
+let apn;
+
+// Load APN certificate and key from S3
+const APN_ENABLED = nconf.get('PUSH_CONFIGS:APN_ENABLED') === 'true';
+
+
+if (APN_ENABLED) {
+  if(nconf.get('NODE_ENV') === 'production') {
+    const S3_BUCKET = nconf.get('S3:bucket');
+    const S3 = new AWS.S3({
+      accessKeyId: nconf.get('S3:accessKeyId'),
+      secretAccessKey: nconf.get('S3:secretAccessKey'),
+    });
+    Bluebird.all([
+      S3.getObject({
+        Bucket: S3_BUCKET,
+        Key: 'apple_apn/cert.pem',
+      }).promise(),
+      S3.getObject({
+        Bucket: S3_BUCKET,
+        Key: 'apple_apn/key.pem',
+      }).promise(),
+    ])
+      .then(([certObj, keyObj]) => {
+        let cert = certObj.Body.toString();
+        let key = keyObj.Body.toString();
+        configureApn(cert, key);
+      });
+  } else {
+    configureApn('../cert.pem', '../key.pem');
+  }
+}
+
+function configureApn(cert, key) {
+  apn = pushNotify.apn({
+    key,
+    cert,
+  });
+
+  apn.on('error', err => logger.error('APN error', err));
+  apn.on('transmissionError', (errorCode, notification, device) => {
+    logger.error('APN transmissionError', errorCode, notification, device);
+  });
+}
+
+function sendNotifications(users, details = {}) {
```

Comment:
You're using a lot of ES6 features, but this codebase is plain node 4 and I'm not sure everything is supported (arrow functions, default parameters, let/const without "use strict"...